--- wuTestService_impl.cpp	2015-01-16 13:35:56.170394903 +0900
+++ wuTestService_implAfter.cpp	2015-01-16 13:45:26.114408448 +0900
@@ -27,18 +27,12 @@
  */
 void wuTestService_impl::start()
 {
-  // Please insert your code here and remove the following warning pragma
-#ifndef WIN32
-  #warning "Code missing in function <void wuTestService_impl::start()>"
-#endif
+  m_comp->start();
 }
 
 void wuTestService_impl::setObjectV(::CORBA::Double x, ::CORBA::Double y, ::CORBA::Double z, ::CORBA::Double roll, ::CORBA::Double pitch, ::CORBA::Double yaw)
 {
-  // Please insert your code here and remove the following warning pragma
-#ifndef WIN32
-  #warning "Code missing in function <void wuTestService_impl::setObjectV(::CORBA::Double x, ::CORBA::Double y, ::CORBA::Double z, ::CORBA::Double roll, ::CORBA::Double pitch, ::CORBA::Double yaw)>"
-#endif
+  m_comp->setObjectV(x, y, z, roll, pitch, yaw);
 }
 
 void wuTestService_impl::testMove()
